File: Pawn.java
import java.io.*;
class Pawn implements Serializable {
    int /* 0-3 */ id;
    String color;
    int location;
    Pawn (int id, String color, int location) {
        this.id=id;
        this.color=color;
        this.location = location;
    }
    
    public boolean equals(Pawn p) {
        return (this.id == p.id && this.color.equals(p.color));
    }
}

File: python/Pawn.py
"""
Implements the Pawn class
"""

#Global

#Local

class Pawn:
    def __init__(self, p_id, color, loc):
        """Takes integer id, string color, location id"""
        self.id = p_id
        self.color = color
        self.location = loc

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.id == other.id and self.color == other.color and self.location == other.location
        return False

if __name__ == "__main__":
    print("the Pawn class")
